### ISSUE TITLE
Error: raise KeyringLocked("Failed to unlock the collection!")

### ISSUE THREAD

**Issue by xiachenrui** [python-poetry/poetry#6640]
- **Poetry version**:  1.2.1
- **Python version**:  3.8.0
- **OS version and name**:  Ubuntu 20.04  
- **pyproject.toml**:  https://gist.github.com/xiachenrui/179c84afd92843effae58ae7c77d3544

- [ ] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) stable Poetry version, installed using a recommended method.
- [ ] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [ ] I have consulted the [FAQ](https://python-poetry.org/docs/faq/) and [blog](https://python-poetry.org/blog/) for any relevant entries or release notes.
- [ ] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option) and have included the output below.

## Issue

Project environment contains an empty path in sys_path, ignoring.
Updating dependencies
Resolving dependencies...
   1: fact: dragon is 0.1.0
   1: derived: dragon
[urllib3.connectionpool] Starting new HTTPS connection (1): download.pytorch.org:443
[urllib3.connectionpool] https://download.pytorch.org:443 "GET /whl/cpu/torch-1.11.0%2Bcpu-cp38-cp38-linux_x86_64.whl HTTP/1.1" 200 169170502
[urllib3.connectionpool] Starting new HTTPS connection (1): data.pyg.org:443
[urllib3.connectionpool] https://data.pyg.org:443 "GET /whl/torch-1.11.0%2Bcpu/torch_scatter-2.0.9-cp38-cp38-linux_x86_64.whl HTTP/1.1" 200 292023
[urllib3.connectionpool] Starting new HTTPS connection (1): data.pyg.org:443
[urllib3.connectionpool] https://data.pyg.org:443 "GET /whl/torch-1.11.0%2Bcpu/torch_sparse-0.6.13-cp38-cp38-linux_x86_64.whl HTTP/1.1" 200 635166
[urllib3.connectionpool] Starting new HTTPS connection (1): data.pyg.org:443
[urllib3.connectionpool] https://data.pyg.org:443 "GET /whl/torch-1.11.0%2Bcpu/torch_cluster-1.6.0-cp38-cp38-linux_x86_64.whl HTTP/1.1" 200 317671
   1: fact: dragon depends on numpy (>1.19)
   1: fact: dragon depends on scipy (>1.3)
   1: fact: dragon depends on pandas (>1.1)
   1: fact: dragon depends on matplotlib (>3.1.2)
   1: fact: dragon depends on seaborn (>0.9)
   1: fact: dragon depends on dill (>0.2.3)
   1: fact: dragon depends on tqdm (>4.27)
   1: fact: dragon depends on scikit-learn (>0.21.2)
   1: fact: dragon depends on statsmodels (>0.10)
   1: fact: dragon depends on parse (>1.3.2)
   1: fact: dragon depends on torch (1.11.0+cpu)
   1: fact: dragon depends on torch-scatter (2.0.9)
   1: fact: dragon depends on torch-sparse (0.6.13)
   1: fact: dragon depends on torch-cluster (1.6.0)
   1: fact: dragon depends on torch-geometric (*)
   1: fact: dragon depends on pytorch_lightning (*)
   1: fact: dragon depends on anndata (>0.7)
   1: fact: dragon depends on scanpy (>1.5)
   1: fact: dragon depends on h5py (*)
   1: fact: dragon depends on packaging (*)
   1: fact: dragon depends on joblib (*)
   1: fact: dragon depends on pyyaml (*)
   1: fact: dragon depends on harmony-pytorch (*)
   1: fact: dragon depends on pynvml (*)
   1: fact: dragon depends on sphinx (*)
   1: fact: dragon depends on sphinx-autodoc-typehints (*)
   1: fact: dragon depends on sphinx-copybutton (*)
   1: fact: dragon depends on sphinx-intl (*)
   1: fact: dragon depends on nbsphinx (*)
   1: fact: dragon depends on sphinx-rtd-theme (*)
   1: fact: dragon depends on ipython (*)
   1: fact: dragon depends on jinja2 (*)
   1: fact: dragon depends on myst-parser (*)
   1: fact: dragon depends on pytest (*)
   1: fact: dragon depends on pytest-cov (*)
   1: selecting dragon (0.1.0)
   1: derived: pytest-cov
   1: derived: pytest
   1: derived: myst-parser
   1: derived: jinja2
   1: derived: ipython
   1: derived: sphinx-rtd-theme
   1: derived: nbsphinx
   1: derived: sphinx-intl
   1: derived: sphinx-copybutton
   1: derived: sphinx-autodoc-typehints
   1: derived: sphinx
   1: derived: pynvml
   1: derived: harmony-pytorch
   1: derived: pyyaml
   1: derived: joblib
   1: derived: packaging
   1: derived: h5py
   1: derived: scanpy (>1.5)
   1: derived: anndata (>0.7)
   1: derived: pytorch_lightning
   1: derived: torch-geometric
   1: derived: torch-cluster (1.6.0) @ https://data.pyg.org/whl/torch-1.11.0%2Bcpu/torch_cluster-1.6.0-cp38-cp38-linux_x86_64.whl
   1: derived: torch-sparse (0.6.13) @ https://data.pyg.org/whl/torch-1.11.0%2Bcpu/torch_sparse-0.6.13-cp38-cp38-linux_x86_64.whl
   1: derived: torch-scatter (2.0.9) @ https://data.pyg.org/whl/torch-1.11.0%2Bcpu/torch_scatter-2.0.9-cp38-cp38-linux_x86_64.whl
   1: derived: torch (1.11.0+cpu) @ https://download.pytorch.org/whl/cpu/torch-1.11.0%2Bcpu-cp38-cp38-linux_x86_64.whl
   1: derived: parse (>1.3.2)
   1: derived: statsmodels (>0.10)
   1: derived: scikit-learn (>0.21.2)
   1: derived: tqdm (>4.27)
   1: derived: dill (>0.2.3)
   1: derived: seaborn (>0.9)
   1: derived: matplotlib (>3.1.2)
   1: derived: pandas (>1.1)
   1: derived: scipy (>1.3)
   1: derived: numpy (>1.19)
[keyring.backend] Loading KWallet
[keyring.backend] Loading SecretService
[keyring.backend] Loading Windows
[keyring.backend] Loading chainer
[keyring.backend] Loading libsecret
[keyring.backend] Loading macOS
   1: Version solving took 39.981 seconds.
   1: Tried 1 solutions.

  Stack trace:

  36  ~/.local/share/pypoetry/venv/lib/python3.8/site-packages/cleo/application.py:329 in run
       327│ 
       328│             try:
     → 329│                 exit_code = self._run(io)
       330│             except Exception as e:
       331│                 if not self._catch_exceptions:

  35  ~/.local/share/pypoetry/venv/lib/python3.8/site-packages/poetry/console/application.py:185 in _run
       183│         self._load_plugins(io)
       184│ 
     → 185│         exit_code: int = super()._run(io)
       186│         return exit_code
       187│ 

  34  ~/.local/share/pypoetry/venv/lib/python3.8/site-packages/cleo/application.py:423 in _run
       421│             io.input.set_stream(stream)
       422│ 
     → 423│         exit_code = self._run_command(command, io)
       424│         self._running_command = None
       425│ 

  33  ~/.local/share/pypoetry/venv/lib/python3.8/site-packages/cleo/application.py:465 in _run_command
       463│ 
       464│         if error is not None:
     → 465│             raise error
       466│ 
       467│         return event.exit_code

  32  ~/.local/share/pypoetry/venv/lib/python3.8/site-packages/cleo/application.py:449 in _run_command
       447│ 
       448│             if event.command_should_run():
     → 449│                 exit_code = command.run(io)
       450│             else:
       451│                 exit_code = ConsoleCommandEvent.RETURN_CODE_DISABLED

  31  ~/.local/share/pypoetry/venv/lib/python3.8/site-packages/cleo/commands/base_command.py:119 in run
       117│         io.input.validate()
       118│ 
     → 119│         status_code = self.execute(io)
       120│ 
       121│         if status_code is None:

  30  ~/.local/share/pypoetry/venv/lib/python3.8/site-packages/cleo/commands/command.py:83 in execute
        81│ 
        82│         try:
     →  83│             return self.handle()
        84│         except KeyboardInterrupt:
        85│             return 1

  29  ~/.local/share/pypoetry/venv/lib/python3.8/site-packages/poetry/console/commands/install.py:145 in handle
       143│         self.installer.verbose(self.io.is_verbose())
       144│ 
     → 145│         return_code = self.installer.run()
       146│ 
       147│         if return_code != 0:

  28  ~/.local/share/pypoetry/venv/lib/python3.8/site-packages/poetry/installation/installer.py:114 in run
       112│             self._execute_operations = False
       113│ 
     → 114│         return self._do_install()
       115│ 
       116│     def dry_run(self, dry_run: bool = True) -> Installer:

  27  ~/.local/share/pypoetry/venv/lib/python3.8/site-packages/poetry/installation/installer.py:247 in _do_install
       245│                 source_root=self._env.path.joinpath("src")
       246│             ):
     → 247│                 ops = solver.solve(use_latest=self._whitelist).calculate_operations()
       248│         else:
       249│             self._io.write_line("Installing dependencies from lock file")

  26  ~/.local/share/pypoetry/venv/lib/python3.8/site-packages/poetry/puzzle/solver.py:73 in solve
        71│         with self._provider.progress():
        72│             start = time.time()
     →  73│             packages, depths = self._solve(use_latest=use_latest)
        74│             end = time.time()
        75│ 

  25  ~/.local/share/pypoetry/venv/lib/python3.8/site-packages/poetry/puzzle/solver.py:151 in _solve
       149│ 
       150│         try:
     → 151│             result = resolve_version(
       152│                 self._package, self._provider, locked=locked, use_latest=use_latest
       153│             )

  24  ~/.local/share/pypoetry/venv/lib/python3.8/site-packages/poetry/mixology/__init__.py:24 in resolve_version
        22│     solver = VersionSolver(root, provider, locked=locked, use_latest=use_latest)
        23│ 
     →  24│     return solver.solve()
        25│ 

  23  ~/.local/share/pypoetry/venv/lib/python3.8/site-packages/poetry/mixology/version_solver.py:127 in solve
       125│             while next is not None:
       126│                 self._propagate(next)
     → 127│                 next = self._choose_package_version()
       128│ 
       129│             return self._result()

  22  ~/.local/share/pypoetry/venv/lib/python3.8/site-packages/poetry/mixology/version_solver.py:407 in _choose_package_version
       405│             dependency = unsatisfied[0]
       406│         else:
     → 407│             dependency = min(*unsatisfied, key=_get_min)
       408│ 
       409│         locked = self._get_locked(dependency)

  21  ~/.local/share/pypoetry/venv/lib/python3.8/site-packages/poetry/mixology/version_solver.py:399 in _get_min
       397│                 return (
       398│                     not dependency.marker.is_any(),
     → 399│                     len(self._dependency_cache.search_for(dependency)),
       400│                 )
       401│             except ValueError:

  20  ~/.local/share/pypoetry/venv/lib/python3.8/site-packages/poetry/mixology/version_solver.py:62 in _search_for
        60│         packages = self.cache.get(key)
        61│         if packages is None:
     →  62│             packages = self.provider.search_for(dependency)
        63│         else:
        64│             packages = [

  19  ~/.local/share/pypoetry/venv/lib/python3.8/site-packages/poetry/puzzle/provider.py:289 in search_for
       287│             return PackageCollection(dependency, packages)
       288│ 
     → 289│         packages = self._pool.find_packages(dependency)
       290│ 
       291│         packages.sort(

  18  ~/.local/share/pypoetry/venv/lib/python3.8/site-packages/poetry/repositories/pool.py:181 in find_packages
       179│         packages = []
       180│         for repo in self._repositories:
     → 181│             packages += repo.find_packages(dependency)
       182│ 
       183│         return packages

  17  ~/.local/share/pypoetry/venv/lib/python3.8/site-packages/poetry/repositories/repository.py:46 in find_packages
        44│         ignored_pre_release_packages = []
        45│ 
     →  46│         for package in self._find_packages(dependency.name, constraint):
        47│             if package.yanked and not isinstance(constraint, Version):
        48│                 # PEP 592: yanked files are always ignored, unless they are the only

  16  ~/.local/share/pypoetry/venv/lib/python3.8/site-packages/poetry/repositories/pypi_repository.py:117 in _find_packages
       115│         """
       116│         try:
     → 117│             info = self.get_package_info(name)
       118│         except PackageNotFound:
       119│             self._log(

  15  ~/.local/share/pypoetry/venv/lib/python3.8/site-packages/poetry/repositories/pypi_repository.py:105 in get_package_info
       103│             return self._get_package_info(name)
       104│ 
     → 105│         package_info: dict[str, Any] = self._cache.store("packages").remember_forever(
       106│             name, lambda: self._get_package_info(name)
       107│         )

  14  ~/.local/share/pypoetry/venv/lib/python3.8/site-packages/cachy/repository.py:174 in remember_forever
       172│             return val
       173│ 
     → 174│         val = value(callback)
       175│ 
       176│         self.forever(key, val)

  13  ~/.local/share/pypoetry/venv/lib/python3.8/site-packages/cachy/helpers.py:6 in value
         4│ def value(val):
         5│     if callable(val):
     →   6│         return val()
         7│ 
         8│     return val

  12  ~/.local/share/pypoetry/venv/lib/python3.8/site-packages/poetry/repositories/pypi_repository.py:106 in <lambda>
       104│ 
       105│         package_info: dict[str, Any] = self._cache.store("packages").remember_forever(
     → 106│             name, lambda: self._get_package_info(name)
       107│         )
       108│         return package_info

  11  ~/.local/share/pypoetry/venv/lib/python3.8/site-packages/poetry/repositories/pypi_repository.py:156 in _get_package_info
       154│ 
       155│     def _get_package_info(self, name: NormalizedName) -> dict[str, Any]:
     → 156│         data = self._get(f"pypi/{name}/json")
       157│         if data is None:
       158│             raise PackageNotFound(f"Package [{name}] not found.")

  10  ~/.local/share/pypoetry/venv/lib/python3.8/site-packages/poetry/repositories/pypi_repository.py:244 in _get
       242│     def _get(self, endpoint: str) -> dict[str, Any] | None:
       243│         try:
     → 244│             json_response = self.session.get(
       245│                 self._base_url + endpoint,
       246│                 raise_for_status=False,

   9  ~/.local/share/pypoetry/venv/lib/python3.8/site-packages/poetry/utils/authenticator.py:246 in get
       244│ 
       245│     def get(self, url: str, **kwargs: Any) -> requests.Response:
     → 246│         return self.request("get", url, **kwargs)
       247│ 
       248│     def post(self, url: str, **kwargs: Any) -> requests.Response:

   8  ~/.local/share/pypoetry/venv/lib/python3.8/site-packages/poetry/utils/authenticator.py:187 in request
       185│     ) -> requests.Response:
       186│         request = requests.Request(method, url)
     → 187│         credential = self.get_credentials_for_url(url)
       188│ 
       189│         if credential.username is not None or credential.password is not None:

   7  ~/.local/share/pypoetry/venv/lib/python3.8/site-packages/poetry/utils/authenticator.py:310 in get_credentials_for_url
       308│                 # no credentials were provided in the url, try finding the
       309│                 # best repository configuration
     → 310│                 self._credentials[url] = self._get_credentials_for_url(url)
       311│             else:
       312│                 # Split from the right because that's how urllib.parse.urlsplit()

   6  ~/.local/share/pypoetry/venv/lib/python3.8/site-packages/poetry/utils/authenticator.py:271 in _get_credentials_for_url
       269│ 
       270│         credential = (
     → 271│             self._get_credentials_for_repository(repository=repository)
       272│             if repository is not None
       273│             else HTTPAuthCredential()

   5  ~/.local/share/pypoetry/venv/lib/python3.8/site-packages/poetry/utils/authenticator.py:259 in _get_credentials_for_repository
       257│ 
       258│         if key not in self._credentials:
     → 259│             self._credentials[key] = repository.get_http_credentials(
       260│                 password_manager=self._password_manager, username=username
       261│             )

   4  ~/.local/share/pypoetry/venv/lib/python3.8/site-packages/poetry/utils/authenticator.py:90 in get_http_credentials
        88│         if credential.password is None:
        89│             # fallback to url and netloc based keyring entries
     →  90│             credential = password_manager.keyring.get_credential(
        91│                 self.url, self.netloc, username=credential.username
        92│             )

   3  ~/.local/share/pypoetry/venv/lib/python3.8/site-packages/poetry/utils/password_manager.py:51 in get_credential
        49│ 
        50│         for name in names:
     →  51│             credential = keyring.get_credential(name, username)
        52│             if credential:
        53│                 return HTTPAuthCredential(

   2  ~/.local/share/pypoetry/venv/lib/python3.8/site-packages/keyring/core.py:72 in get_credential
        70│ ) -> typing.Optional[credentials.Credential]:
        71│     """Get a Credential for the specified service."""
     →  72│     return get_keyring().get_credential(service_name, username)
        73│ 
        74│ 

   1  ~/.local/share/pypoetry/venv/lib/python3.8/site-packages/keyring/backends/SecretService.py:112 in get_credential
       110│         scheme = self.schemes[self.scheme]
       111│         query = self._query(service, username)
     → 112│         collection = self.get_preferred_collection()
       113│ 
       114│         with closing(collection.connection):

  KeyringLocked

  Failed to unlock the collection!

  at ~/.local/share/pypoetry/venv/lib/python3.8/site-packages/keyring/backends/SecretService.py:67 in get_preferred_collection
       63│             raise InitError("Failed to create the collection: %s." % e)
       64│         if collection.is_locked():
       65│             collection.unlock()
       66│             if collection.is_locked():  # User dismissed the prompt
    →  67│                 raise KeyringLocked("Failed to unlock the collection!")
       68│         return collection
       69│ 
       70│     def unlock(self, item):
       71│         if hasattr(item, 'unlock'):
